File: dataModule/src/main/java/com/mobnetic/coinguardian/model/currency/CurrenciesSubunits.java
package com.mobnetic.coinguardian.model.currency;

import java.util.HashMap;

import com.mobnetic.coinguardian.model.CurrencySubunit;
import com.mobnetic.coinguardian.model.CurrencySubunitsMap;

public class CurrenciesSubunits {

	public final static HashMap<String, CurrencySubunitsMap> CURRENCIES_SUBUNITS = new HashMap<String, CurrencySubunitsMap>();
	static {
		CURRENCIES_SUBUNITS.put(VirtualCurrency.BTC, new CurrencySubunitsMap(
				new CurrencySubunit(VirtualCurrency.BTC, 1),
				new CurrencySubunit(VirtualCurrency.mBTC, 1000),
				new CurrencySubunit(VirtualCurrency.uBTC, 1000000),
				new CurrencySubunit(VirtualCurrency.Satoshi, 100000000, false)
			)
		);
		CURRENCIES_SUBUNITS.put(VirtualCurrency.LTC, new CurrencySubunitsMap(
				new CurrencySubunit(VirtualCurrency.LTC, 1),
				new CurrencySubunit(VirtualCurrency.mLTC, 1000)
			)
		);
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/model/currency/CurrenciesSubunits.kt
package com.aneonex.bitcoinchecker.datamodule.model.currency

import com.aneonex.bitcoinchecker.datamodule.model.CurrencySubunit
import com.aneonex.bitcoinchecker.datamodule.model.CurrencySubunitsMap

object CurrenciesSubunits {
    val CURRENCIES_SUBUNITS = mapOf(
        VirtualCurrency.BTC to CurrencySubunitsMap(
            CurrencySubunit(VirtualCurrency.BTC, 1),
            CurrencySubunit(VirtualCurrency.mBTC, 1000),
//                CurrencySubunit(VirtualCurrency.uBTC, 1000000),
            CurrencySubunit(VirtualCurrency.Satoshi, 100000000, false)
        ),
        VirtualCurrency.LTC to CurrencySubunitsMap(
            CurrencySubunit(VirtualCurrency.LTC, 1),
            CurrencySubunit(VirtualCurrency.mLTC, 1000)
        )
    )
}